Lib/concurrent/futures/process.py

```diff
@@ -713,15 +719,27 @@ def _adjust_process_count(self):
 
         process_count = len(self._processes)
         if process_count < self._max_workers:
-            p = self._mp_context.Process(
-                target=_process_worker,
-                args=(self._call_queue,
-                      self._result_queue,
-                      self._initializer,
-                      self._initargs,
-                      self._max_tasks_per_child))
-            p.start()
-            self._processes[p.pid] = p
+            assert self._safe_to_dynamically_spawn_children or not self._executor_manager_thread, 'https://github.com/python/cpython/issues/90622'
```

Comment:
Hmm, I'm worried this assert may hit and break existing code, can we remove it?

Reply:
I'll definitely remove it entirely from any backport.  I'd _like_ to see if it trips anything up in 3.11beta1 before deciding to remove it though. It _should_ only come up if people are calling this non-public API themselves (subclassing?). Perhaps turning it into a warning for 3.11 is safer?

Reply:
assertion disabled. comment added. we still have a bug here, though it should be a rare one: if a 'fork' mp_context worker dies, we'll spawn (fork) a replacement despite knowing that a thread is running.